docs/architecture/core/gateway.puml

```diff
@@ -0,0 +1,64 @@
+@startuml gateway
+title Gateway
```

Comment:
The amount of arrows from/to the `Message*` contracts is overwhelming for me. Maybe for `Root` we can add a note that it has superuser power to remove some arrows. For adapters, I would prefer a single arrow from the `adapters` block to the Gatway over each single adapter.